src/openms/source/METADATA/SpectrumMetaDataLookup.cpp

```diff
@@ -246,6 +246,40 @@ namespace OpenMS
     return success;
   }
 
+  bool SpectrumMetaDataLookup::addMissingIMToPeptideIDs(vector<PeptideIdentification>& peptides, const String& filename,
+  bool stop_on_error)
+  {
+    PeakMap exp;
+    SpectrumLookup lookup;
+    bool success = true;
+    for (auto& pep : peptides)
+    {
+      if (lookup.empty())
+      {
+        FileHandler fh;
+        auto opts = fh.getOptions();
+        // speed up reading. We do not need the actual peaks in the spectra
+        opts.setFillData(false);
+        opts.setSkipXMLChecks(true);
+        fh.setOptions(opts);
+        fh.loadExperiment(filename, exp, {FileTypes::MZXML, FileTypes::MZML, FileTypes::MZDATA, FileTypes::MGF}, OpenMS::ProgressLogger::NONE, true, true);
```

Comment:
does MGF store IM?
We should have a 'database' somewhere which tells you which formats support what data...
Ideally, each format's handler can tell on its own...

Reply:
I don't think so.
But I guess if we change the signature we don't need to deal with that here.